homeassistant/components/netatmo/cover.py

```diff
@@ -0,0 +1,144 @@
+
+
+class NetatmoCover(NetatmoBase, CoverEntity):
+    """Representation of a Netatmo cover device."""
+
+    def __init__(self, netatmo_device: NetatmoDevice) -> None:
+        """Initialize the Netatmo device."""
+        CoverEntity.__init__(self)
+        super().__init__(netatmo_device.data_handler)
+        self.optimistic = True
+
+        self._cover = cast(NaModules.Shutter, netatmo_device.device)
+
+        self._id = self._cover.entity_id
+        self._attr_name = self._device_name = self._cover.name
+        self._model = self._cover.device_type
+        self._config_url = CONF_URL_CONTROL
+
+        self._home_id = self._cover.home.entity_id
+        self._closed: bool | None = None
+        self._attr_is_closed = None
+
+        self._signal_name = f"{HOME}-{self._home_id}"
+        self._publishers.extend(
+            [
+                {
+                    "name": HOME,
+                    "home_id": self._home_id,
+                    SIGNAL_NAME: self._signal_name,
+                },
+            ]
+        )
+        self._attr_unique_id = f"{self._id}-{self._model}"
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        supported_features = 0
+        supported_features |= CoverEntityFeature.OPEN
+        supported_features |= CoverEntityFeature.CLOSE
+        supported_features |= CoverEntityFeature.STOP
+        supported_features |= CoverEntityFeature.SET_POSITION
+
+        return supported_features
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        self._attr_is_closing = True
+        self.async_write_ha_state()
+        try:
+            await self._cover.async_close()
+            if self.optimistic:
+                self._attr_is_closed = True
+        finally:
+            self._attr_is_closing = None
+            self.async_write_ha_state()
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        self._attr_is_opening = True
+        self.async_write_ha_state()
+        try:
+            await self._cover.async_open()
+            if self.optimistic:
+                self._attr_is_closed = False
+        finally:
+            self._attr_is_opening = None
+            self.async_write_ha_state()
+
+    async def async_stop_cover(self, **kwargs: Any) -> None:
+        """Stop the cover."""
+        await self._cover.async_stop()
+
+        if self.optimistic:
+            if self._attr_is_closing:
+                self._attr_is_closed = True
+            elif self._attr_is_opening:
+                self._attr_is_closed = False
+
+            self._attr_is_closing = None
+            self._attr_is_opening = None
+
+        self.async_write_ha_state()
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Move the cover shutter to a specific position."""
+        await self._cover.async_set_target_position(kwargs[ATTR_POSITION])
```

Comment:
Its an async library right? Adding async in method names is typically home assistant, no need to reflect this in the library

Reply:
That is true, but right now the library actually provides sync and async methods in many places. This will be the next major iteration of pyatmo.